### PR TITLE
Use FetchContent() for googletest

### PR DESCRIPTION
**Pull request by havogt**
Use FetchContent to make googletest available.

### CODE REVIEW

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch icgrid

**Comment by havogt** [comment]
TODO: run perftest and mpi test if the other checks passed

**Comment by havogt** [comment]
launch perftest

**Comment by havogt** [comment]
launch mpistrgrid

File: cmake/testing.cmake [hunk 1]
```diff
@@ -17,8 +17,19 @@ target_link_libraries( GridToolsTest INTERFACE Threads::Threads)
...
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.8.1
+)
+FetchContent_GetProperties(googletest)
+if(NOT googletest_POPULATED)
+  FetchContent_Populate(googletest)
+  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
+endif()
```
Comment (lukasm91) [inline cmake/testing.cmake]: Lines 28-32 are equivalent to

```
FetchContent_MakeAvailable(googletest)
```
https://github.com/Kitware/CMake/blob/master/Modules/FetchContent.cmake#L1041

File: cmake/testing.cmake [hunk 2]
```diff
@@ -17,8 +17,19 @@ target_link_libraries( GridToolsTest INTERFACE Threads::Threads)
 include(workaround_threads)
 _fix_threads_flags()
 
+include(FetchContent)
 option(INSTALL_GTEST OFF)
-add_subdirectory(./tools/googletest)
+mark_as_advanced(INSTALL_GTEST)
```
Comment (lukasm91) [inline cmake/testing.cmake]: Maybe you can add a TODO or make it even an explicit workaround before 3.13: INSTALL_GTEST doesn't need to be a cached variable when CMake >= 3.13:

https://cmake.org/cmake/help/v3.13/command/option.html

File: cmake/testing.cmake (hunk 1 above)
Comment (havogt) [reply cmake/testing.cmake]: Only available with CMake 3.14.

File: cmake/testing.cmake [hunk 3]
```diff
@@ -17,8 +17,19 @@ target_link_libraries( GridToolsTest INTERFACE Threads::Threads)
 include(workaround_threads)
 _fix_threads_flags()
 
+include(FetchContent)
 option(INSTALL_GTEST OFF)
```
Comment (havogt) [inline cmake/testing.cmake]:
```suggestion
option(INSTALL_GTEST OFF) #TODO replace with set(INSTALL_GTEST OFF) with CMake >= 3.13
```

File: cmake/testing.cmake [hunk 4]
```diff
@@ -17,8 +17,19 @@ target_link_libraries( GridToolsTest INTERFACE Threads::Threads)
...
-add_subdirectory(./tools/googletest)
+include(FetchContent)
+option(INSTALL_GTEST OFF) #TODO replace with set(INSTALL_GTEST OFF) with CMake >= 3.13
+mark_as_advanced(INSTALL_GTEST)
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.8.1
+)
+FetchContent_GetProperties(googletest)
```
Comment (havogt) [inline cmake/testing.cmake]:
```suggestion
# TODO Replace the next 5 lines with `FetchContent_MakeAvailable(googletest)` once we upgrade to CMake 3.14+.
FetchContent_GetProperties(googletest)
```

File: cmake/testing.cmake (hunk 1 above)
Comment (lukasm91) [reply cmake/testing.cmake]: Didn't see that.

**Comment by lukasm91** [comment]
launch strgrid

**Pull request merged**